lib/plugins/output/elasticsearch.js

```diff
@@ -194,12 +194,18 @@ OutputElasticsearch.prototype.eventHandler = function (data, context) {
 
   if (index) {
     index = applyDateFormatToIndex(index, data)
+    if (data._index != undefined) {
+      data._index = index
+    }
     this.indexData(index, data._type || 'logs', data, config, logsReceiverUrl)
   }
 
   if (context.logsReceivers && context.logsReceivers.length > 0) {
     context.logsReceivers.forEach(receiver => {
       const receiverIndex = applyDateFormatToIndex(receiver.index, data)
+      if (data._index != undefined) {
```

Comment:
To align with our linting rules, can you edit this to use strict inequality?

```suggestion
      if (data._index !== undefined) {
```